static/src/javascripts/projects/common/modules/ui/cookiesBanner.js

```diff
@@ -20,24 +20,27 @@ define([
     /**
      * Rules:
      *
-     * EU visitors only
+     * UK / INT edition readers only
      * Never seen the cookie message before
      * Show once only
      * Show only on FIRST page view
      * Persist close state
      */
     function init() {
-        var EU_COOKIE_MSG = 'GU_EU_MSG',
-            euMessageCookie = cookies.get(EU_COOKIE_MSG);
-
-        if (euMessageCookie && euMessageCookie === 'unseen') {
-            var link = 'https://www.theguardian.com/info/cookies',
-                txt = 'Welcome to the Guardian. This site uses cookies, read our policy <a href="' + link + '" class="cookie-message__link">here</a>',
-                opts = {important: true},
-                cookieLifeDays = 365,
-                msg = new Message('cookies');
-            msg.show(txt, opts);
-            cookies.add(EU_COOKIE_MSG, 'seen', cookieLifeDays);
+        if (config.switches.euCookieMsg) {
+            if (config.page.edition === 'UK' || config.page.edition === 'INT') {
+                var EU_COOKIE_MSG = 'GU_EU_MSG',
+                    euMessageCookie = cookies.get(EU_COOKIE_MSG);
+                if (!euMessageCookie || euMessageCookie != 'seen') {
+                    var link = 'https://www.theguardian.com/info/cookies',
```

Comment:
this has been done a couple of times, and from my memory I think the summary was that it's mitigated by functions being kept fairly focussed (massive functions have other problems not just scoping issues), and it's also working around a surprising for some effect of javascript where the vars are declared at that point even if you write them in the code later.
However I'm not in either camp (other than the keep functions short camp)